design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,306 @@
+
+#### Releases
+- **Alpha**:
+  An initial release of the feature for experimental purposes.
+  Recommended for non-production usages, evaluation or testing.
+
+  The API is considered unstable and may change significantly.
+
+  The period in which a feature can remain in Alpha is limited,
+  assuring features are not piling up without control.
+
+  The feature functionality is controlled using a Feature-Gate (FG) during
+  runtime. It must be specified for the feature to be active.
+
+- **Beta**:
+  The first release that can be evaluated with care in production.
+  Acting as a pre-release, its main objective is to collect feedback
+  from users to assure its usefulness and readiness for graduation.
+  If there is no confidence of usage or usefulness, it may remain in
+  this stage for some time.
+  However, the period in which a feature can remain in Beta is limited,
+  assuring features are not piling up without control.
+
+  The API is considered stable with care no not break backward compatibility
+  with previous beta releases.
+  This implies that fields may only be added during this stage,
+  not removed or renamed.
+
+  The feature functionality is controlled using a Feature-Gate (FG) during
+  runtime. It must be specified for the feature to be active.
+
+- **GA**:
+  The feature graduated to general-availability (GA) and is now part of
+  the core features.
+
+  The feature functionality is no longer controlled by a FG.
+
+> **Warning**: A Feature Gate flag is solely intended to control
+> feature lifecycle. It should not be confused and used as a cluster
+> configurable enablement of the functionality.
+> In cases where the cluster admin should control a functionality,
+> regardless to the feature stage, dedicated configuration field/s
+> should be included.
+
+#### Removal
+When and if a feature is targeted for deprecation and retirement,
+it needs to pass a deprecation process, depending on its current
+release stage (Alpha, Beta, GA).
+
+For more details, see [here](#deprecation-and-removal).
+
+#### Release Stage Transition Table
+The following table summarized the different release stages with their
+transition requirements and restrictions.
+
+| Stage          | Period range    | F.Gate | Removal Availability       |
+|----------------|-----------------|--------|----------------------------|
+| Alpha          | 1 to 2 releases | YES    | Between **minor** releases |
+| Beta           | 1 to 3 releases | YES    | Between **minor** releases |
+| GA             | -               | NO     | Between **major** releases |
+
+Through Alpha and Beta feature releases, a FG must be set in order
+for the feature to function.
+By default, no FG is specified, therefore the feature is by disabled.
+
+If a feature is not able to transition to the next stage in the defined period,
+it should be removed automatically.
+
+> **Note**: Exceptions to the period range may apply
+> if 2/3 of active maintainers come to agreement to prolong
```

Comment:
In future we should delegate this to a sig.

Reply:
@EdDev At the exceptions section below you wrote:
```
maintainers (also known as "approvers").
```

Do you mind adding this to here as well? I wasn't (and still not) sure what's the different between maintainers and approvers, but either way it could be useful to clarify it here as well.

Reply:
@iholder101 
These links should cover these terms
https://github.com/kubevirt/community/blob/main/GOVERNANCE.md
https://github.com/kubevirt/community/blob/main/membership_policy.md

Reply:
I agree that 2/3 of active maintainers sounds very strange to me. Are the active maintainers even aware of this?

Reply:
I agree that 2/3 of active maintainers sounds very strange to me. Are the active maintainers even aware of this?

Reply:
The term maintainers here is confusing i would prefer using "approvers" instead

Reply:
>The term maintainers here is confusing i would prefer using "approvers" instead

:+1: 
Just to be consistent with the exception section below I think it would be best

Reply:
One is a project/repo maintainer and the other an org one.
The fact the term “approve” is used in the k8s ecosystem does not mean you should not be familiar with the more general term “maintainer”.

I am a bit surprised to be honest from the discussion around this topic.
AFAIK, open source projects have maintainers, in some projects they are called approvers because the action is called `approve` (vs other projects where it is `+2`).

I do not feel comfortable to be so defensive when using common terms like this and re-clarify the meaning.
But, if this is the only thing left that blocks this proposal to get in, I will add the same clarification.

https://en.m.wikipedia.org/wiki/Software_maintainer

Reply:
It is not a blocker from my side, just thought it would be clearer.
As long as it's written in the exception section below it's resolved from my side.